SurveyAppTests/Tests/Services/Network/API/UserSessionTests.swift

```diff
@@ -0,0 +1,55 @@
+//
+//  UserSessionTests.swift
+//  SurveyAppTests
+//
+//  Created by Chananchida F. on 6/17/21.
+//
+
+import Quick
+import Nimble
+
+@testable import SurveyApp
+
+class UserSessionTests: QuickSpec {
+    
+    override func spec() {
+        
+        var keychain: KeychainStorageProtocol!
+        var userSession: UserSession!
+        
+        let sampleUserCredential: UserCredential = JSON.AuthenticationService.sampleUserCredential.decoded()
+    
+        describe("UserSession") {
+            
+            beforeEach {
+                keychain = KeychainStorage.default
+                userSession = UserSession(keychain: keychain)
+                UserSessionProvider.shared.userSession = userSession
+                userSession.userCredential = sampleUserCredential
+            }
+            
+            afterEach {
+                UserSessionProvider.shared.clearCredentials()
+            }
+            
+            context("when setting user credentials") {
+                it("should set the access and refresh token in UserSession") {
+                    expect(userSession.userCredential?.accessToken) == "sampleAccessToken"
+                    expect(userSession.userCredential?.refreshToken) == "sampleRefreshToken"
```

Comment:
I think we should check `expect` with the object we are dummying
```suggestion
                    expect(userSession.userCredential?.accessToken) ==  sampleUserCredential.sampleAccessToken
                    expect(userSession.userCredential?.refreshToken) == sampleUserCredential .sampleRefreshToken
```

Reply:
When `sampleUserCredential` decoded it will be `UserCredential` object.

So should I update the code to be like this?
```Swift
expect(userSession.userCredential?.accessToken) == sampleUserCredential.accessToken
```

Reply:
Yes, thank you.

Reply:
@blyscuit I updated the code please help check 🙇‍♀️